lib/Serialization/Serialization.cpp

```diff
@@ -4494,9 +4494,12 @@ class Serializer::DeclSerializer : public DeclVisitor<DeclSerializer> {
     uint8_t rawAccessLevel = getRawStableAccessLevel(ctor->getFormalAccess());
 
     bool firstTimeRequired = ctor->isRequired();
-    if (auto *overridden = ctor->getOverriddenDecl())
+    auto *overridden = ctor->getOverriddenDecl();
+    if (overridden) {
       if (firstTimeRequired && overridden->isRequired())
         firstTimeRequired = false;
+    }
+    bool overriddenAffectsABI = overriddenDeclAffectsABI(ctor, overridden);
```

Comment:
Is the relationship actually useful if it doesn't?

Reply:
That's a good question. For non-resilient modules it could be used by optimizations. For resilient modules in most cases it should now be dropped by deserialization safety for the public-override-internal case, and the clang case is there more as tolerance for poor modularization and such I believe. I think we could try always dropping it when reading from a resilient module.